pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -668,72 +669,13 @@ func (r *TeamRoleBindingReconciler) listClusters(ctx context.Context, trb *green
 	return clusters, nil
 }
 
-// initTeamRoleBindingStatus ensures that all required conditions are present in the TeamRoleBinding's Status
-func initTeamRoleBindingStatus(trb *greenhousev1alpha1.TeamRoleBinding) greenhousev1alpha1.TeamRoleBindingStatus {
-	status := trb.Status.DeepCopy()
-	for _, ct := range exposedConditions {
-		if status.GetConditionByType(ct) == nil {
-			status.SetConditions(greenhousev1alpha1.UnknownCondition(ct, "", ""))
-		}
-	}
-	return *status
-}
```

Comment:
The idea with the `exposedConditions` was that this `initTRBStatus(..)` would set all Conditions to `unknown` to ensure the status is complete, even if the controller errors out early in the reconciliation loop. 
This is to follow the general guidance for [kube-api conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
```text
Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown. This allows other components in the system to know that the condition exists and the controller is making progress on reconciling that resource.
```